File: solutions/1431. Kids With the Greatest Number of Candies/1431.cs
// Time complexity: O(n)
// Space complexity: O(n)
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        List<bool> ans = new List<bool>();
        int maxCandy = 0;

        foreach(int candy in candies) {
            maxCandy = Math.Max(candy, maxCandy);
        }

        foreach(int candy in candies) {
            ans.Add(candy + extraCandies >= maxCandy);
        }

        return ans;
    }    
}

File: solutions/1431. Kids With the Greatest Number of Candies/1431.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
  def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
    maxCandy = max(candies)
    return [candy + extraCandies >= maxCandy for candy in candies]